kernel/src/sched.rs

```diff
@@ -1183,46 +1180,44 @@ impl Kernel {
             } => {
                 let res = platform.with_driver(driver_number, |driver| match driver {
                     Some(d) => {
-                        // Try to create an appropriate [`ReadOnlyAppSlice`].
+                        // Try to create an appropriate [`ReadOnlyProcessBuffer`].
                         // This method will ensure that the memory in question
                         // is located in the process-accessible memory space.
-                        //
-                        // TODO: Enforce anti buffer-aliasing guarantees to avoid
-                        // undefined behavior in Rust.
-                        match process.build_readonly_appslice(allow_address, allow_size) {
-                            Ok(appslice) => {
-                                // Creating the [`ReadOnlyAppSlice`] worked,
+                        match process.build_readonly_process_buffer(allow_address, allow_size) {
+                            Ok(ro_pbuf) => {
+                                // Creating the [`ReadOnlyProcessBuffer`] worked,
                                 // provide it to the capsule.
                                 match d.allow_readonly(
                                     process.processid(),
                                     subdriver_number,
-                                    appslice,
+                                    ro_pbuf,
                                 ) {
-                                    Ok(returned_appslice) => {
+                                    Ok(returned_pbuf) => {
                                         // The capsule has accepted the allow
                                         // operation. Pass the previous buffer
                                         // information back to the process.
                                         //
-                                        // TODO: Prevent swapping of AppSlices by
+                                        // TODO: Prevent swapping of ProcessBuffers by
```

Comment:
What is the status of these TODOs?

Reply:
Citing @hudson-ayers 

> I think it is ok if we release Tock 2.0 without addressing AppSlice swapping, and then just address it alongside any proposal to allow moving an app brk down.

I'd still say that we might want to address this at some point, when convenient, possibly when we arrive at a `trait Process` implementation which supports moving `app_brk` down. Hence, I'd vouch to deliberately leave this in.

Reply:
Gotcha, do we have a decent reference we can point to about that todo, like an issue? I'm worried we'll forget what the intent is and why we didn't do it for 2.0.

Reply:
I'll point to this issue, specifically your comment about this? I think that's a pretty decent reference.

Reply:
Done. Feel free to resolve if this works for you.

Reply:
As discussed on the call, I'll remove this comment and create a feature-request issue, plus PR an update to TRD104.